### ISSUE TITLE
units of nbodykit.transform.SkyToCartesian

### ISSUE THREAD

**Issue by SandyYuan** [bccp/nbodykit#607]
The documentation says that the returned coordinates are in Mpc / h, but the results look like they are in Mpc instead.

**Comment by rainwoodman** [comment]
Did you check if the cosmology object has h=1. If so then Mpc/h and Mpc are the same.

**Comment by SandyYuan** [comment]
Yeah I passed it custom cosmology with H0 = 67.26. I assume internally the code converts that to an h = 0.6726.

**Comment by rainwoodman** [comment]
Indeed there is a conversion.

This is strange. Could be a serious bug or a regression -- although
unlikely. Last time we checked this was definitely Mpc/h.

We shall figure it out and add a test case, to make sure it never goes bad
again.

Could you post a code snippet & what you expect & what you get?

Thanks!

**Comment by SandyYuan** [comment]
code:
`
from nbodykit.transform import SkyToCartesian

from astropy.cosmology import FlatLambdaCDM
cosmo = FlatLambdaCDM(H0= 67.26 , Om0=0.316)

SkyToCartesian([0], [0], [1], cosmo, 
        observer=[0, 0, 0], degrees=True, frame='icrs').compute()
`
It gives 

array([[ 3406.04607083,     0.        ,     0.        ]])

That is the comoving distance in Mpc. To check that, we can simply do 
`
cosmo.comoving_distance(1)
`
which gives 

<Quantity 3406.046070834424 Mpc>

**Comment by nickhand** [comment]
You'll get back the coordinates in whatever units the input cosmo.comoving_distance() returns. In the case of nbodykit's Cosmology object, the comoving_distance() returns distances in units of Mpc/h since it's built on CLASS. The function

The function was intended to be used with nbodykit's cosmology object, but we should update the documentation to better reflect what is actually happening.

**Comment by NiallJeffrey** [comment]
I have also found this problem. It seems that CartesianToSky is taking only scalar values (ignoring units) and assuming the units are Mpc - not Mpc/h as the documentation says: https://nbodykit.readthedocs.io/en/latest/api/_autosummary/nbodykit.transform.html#nbodykit.transform.CartesianToSky

**Comment by rainwoodman** [comment]
What type is the cosmology object did you provide to the function?

**Comment by sbird** [comment]
Closing.
